clientv3/maintenance.go

```diff
@@ -61,6 +61,10 @@ func NewMaintenance(c *Client) Maintenance {
 	return &maintenance{c: c, remote: pb.NewMaintenanceClient(c.conn)}
 }
 
+func NewMaintenanceFromMaintenanceClient(remote pb.MaintenanceClient) Maintenance {
+	return &maintenance{remote: remote}
```

Comment:
seems like the `c *Client` in `maintenance` be replaced with `dial func(endpoint string) (pb.MaintenanceClient, error)` so this won't crash out for the v3client case? NewMaintenanceFromMaintenanceClient could always `remote` / some error if it's a non-local endpoint.